combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/ProbeBleDevice.kt

```diff
@@ -146,9 +146,39 @@ internal class ProbeBleDevice (
     }
 
     override suspend fun readSerialNumber() = uart.readSerialNumber()
+
     override suspend fun readFirmwareVersion() = uart.readFirmwareVersion()
+    fun readProbeFirmwareVersion(callback: (FirmwareVersion) -> Unit) {
+        uart.owner.lifecycleScope.launch {
+            readFirmwareVersion()
+        }.invokeOnCompletion {
+            deviceInfoFirmwareVersion?.let {
```

Comment:
Should we be calling back with a default/invalid value if `deviceInfoFirmwareVersion` is `null`? As-is, the caller won't ever know if the read failed.

Looking through the code it doesn't appear that this is much of an issue, so we can probably rack it up to tech debt.